### PR TITLE
Fetch image

### PR DESCRIPTION
**Pull request by gguijarro-c-chwy**

### CODE REVIEW

**Comment by gguijarro-c-chwy** [comment]
@n13  I need ur help here. I spent way too long already trying to figure out this Image stuff for the Daos. 
Im about to just add the images to our firebase and be doe with it.

File: lib/ui/profile/components/dao_widget.dart [hunk 1]
```diff
@@ -13,54 +15,71 @@ class DaoWidget extends StatelessWidget {
...
+                  Row(
+                    crossAxisAlignment: CrossAxisAlignment.center,
+                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                    children: [
+                      Row(
+                        children: [
+                          const Icon(Icons.calendar_month),
+                          const SizedBox(width: 4),
+                          Text(
+                            'The date here',
```
Comment (n13) [inline lib/ui/profile/components/dao_widget.dart:66]: Should this say "the date here"?

File: lib/ui/profile/components/dao_widget.dart [hunk 2]
```diff
@@ -13,54 +15,71 @@ class DaoWidget extends StatelessWidget {
...
+                    mainAxisAlignment: MainAxisAlignment.start,
+                    children: [
+                      const Icon(Icons.ac_unit, size: 48),
+                      const SizedBox(width: 8),
+                      Column(
+                        crossAxisAlignment: CrossAxisAlignment.start,
+                        children: [
+                          Text(dao.settingsDaoTitle, style: context.hyphaTextTheme.smallTitles),
+                          Text(
+                            'dao.hypha.earth/${dao.settingsDaoUrl}',
```
Comment (n13) [inline lib/ui/profile/components/dao_widget.dart:47]: I thought this field was called short URL or something

File: lib/core/network/ipfs/ipfs_manager.dart [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import 'package:ipfs_client_flutter/ipfs_client_flutter.dart';
+
+class IPFSManager {
+  final apiKey = 'apiKey';
+  final apiKeySecret = 'apiKeySecret';
+  final serverUlr = 'https://ipfs.infura.io:5001';
```
Comment (n13) [inline lib/core/network/ipfs/ipfs_manager.dart:6]: Ok... guess we need secret - let's load it from firebase?! At least it won't be in the code then..

File: lib/core/network/ipfs/ipfs_manager.dart [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  late IpfsClient ipfsClient;
+
+  IPFSManager() {
+    final apiKeySecret = dotenv.env['IPFS_API_KEY_SECRET'];
+    ipfsClient = IpfsClient(url: serverUlr, authorizationToken: apiKeySecret);
+  }
+
+  Future getImage(String imageToken) async {
+    final response = await ipfsClient.read(dir: imageToken);
```
Comment (gguijarro-c-chwy) [inline lib/core/network/ipfs/ipfs_manager.dart:14]: @n13  this doesnt work. it fail. can you take a look when you get some time pls.

Comment (n13) [reply lib/core/network/ipfs/ipfs_manager.dart:14]: Hmmm

File: lib/ui/profile/components/dao_widget.dart (hunk 1 above)
Comment (gguijarro-c-chwy) [reply lib/ui/profile/components/dao_widget.dart:66]: We dont have this field in model response. 

Is this data available for every dao? 

Do you have an account with multiple daos i can use?

Comment (gguijarro-c-chwy) [reply lib/ui/profile/components/dao_widget.dart:66]: nvm Leo will let me use hers

Comment (gguijarro-c-chwy) [reply lib/ui/profile/components/dao_widget.dart:66]: image

Comment (gguijarro-c-chwy) [reply lib/ui/profile/components/dao_widget.dart:66]: We dont have this date

Comment (gguijarro-c-chwy) [reply lib/ui/profile/components/dao_widget.dart:66]: Unless we need to add more params to the query. Its a graphQl query that you created.

Comment (n13) [reply lib/ui/profile/components/dao_widget.dart:66]: We don't have this data, let's not show it

**Comment by n13** [comment]
Ok so this item is in limbo because the IPFS access isn't working? 

Or is it working now with our .env file?

**Comment by n13** [comment]
Current status: Not working. 

IPFS access isn't working yet

**Comment by n13** [comment]
Some examples of IPFS images

```
flutter: ║                 memberof: [
flutter: ║                    {
flutter: ║                         docId: "29983",
flutter: ║                         details_daoName_n: "hypha",
flutter: ║                         settings: [
flutter: ║                            {
flutter: ║                                 settings_daoTitle_s: "Hypha DAO",
flutter: ║                                 settings_isHypha_i: 1,
flutter: ║                                 "QmV3KmaoqCCXuCDvHzYWS9Jg3RfjrDTQSXK1e7453qfSRS:svg"
flutter: ║                                 settings_daoUrl_s: "hypha"
flutter: ║                            }
flutter: ║                         ]
flutter: ║                    },
flutter: ║                    {
flutter: ║                         docId: "31631",
flutter: ║                         details_daoName_n: "lightleaders",
flutter: ║                         settings: [
flutter: ║                            {
flutter: ║                                 settings_daoTitle_s: "The Lightleaders",
flutter: ║                                 settings_isHypha_i: null,
flutter: ║                                 "QmUyxopcxMLMPmo4UWDrE5d4P3yka4AwWJH4MASxpcWeEy:png"
flutter: ║                                 settings_daoUrl_s: "thelightleaders"
flutter: ║                            }
flutter: ║                         ]
flutter: ║                    },
flutter: ║                    {
flutter: ║                         docId: "31991",
flutter: ║                         details_daoName_n: "demoxdaox",
flutter: ║                         settings: [
flutter: ║                            {
flutter: ║                                 settings_daoTitle_s: "Demo DAO",
flutter: ║                                 settings_isHypha_i: null,
flutter: ║                                 "QmNMv449YPqbWVncve9eyNZfgi2RKwujCDmeUfBhjyX9j4:jpeg"
flutter: ║                                 settings_daoUrl_s: "demoxdaox"
flutter: ║                            }
flutter: ║                         ]
flutter: ║                    },
flutter: ║                    {
flutter: ║                         docId: "43454",
flutter: ║                         details_daoName_n: "theflow",
flutter: ║                         settings: [
flutter: ║                            {
flutter: ║                                 settings_daoTitle_s: "THE FLOW",
flutter: ║                                 settings_isHypha_i: null,
flutter: ║                                 "QmbyW3yY7jTjUi6XCkSy5GMsPDtonXBQ6e5nTFj3Dh6P9z:jpeg"
flutter: ║                                 settings_daoUrl_s: "the-flow"
flutter: ║                            }
flutter: ║                         ]
flutter: ║                    }
flutter: ║                 ],

```

**Comment by n13** [comment]
Hmm this works

https://ipfs.io/ipfs/QmNMv449YPqbWVncve9eyNZfgi2RKwujCDmeUfBhjyX9j4

image

**Comment by n13** [comment]
https://ipfs.io/ipfs/QmV3KmaoqCCXuCDvHzYWS9Jg3RfjrDTQSXK1e7453qfSRS

this also works... in the browser.. it's an svg

**Comment by n13** [comment]
Ok all done - it's working so I am leaving it with you to put the files in the right places and so on... I made a new image widget IPFSImage for showing the URLs. 

image

**Comment by n13** [comment]
The linter complains there's no .env file... a conundrum...

**Pull request merged**